app/src/main/java/com/github/multimatum_team/multimatum/AddDeadlineActivity.kt

```diff
@@ -78,20 +92,38 @@ class AddDeadlineActivity : AppCompatActivity() {
             TimePickerDialog.OnTimeSetListener { _, hoursOfDay, minutes ->
                 selectedDate = LocalDateTime.of(
                     selectedDate.year, selectedDate.monthValue, selectedDate.dayOfMonth,
-                    hoursOfDay, minutes)
+                    hoursOfDay, minutes
+                )
 
                 textTime.text = selectedDate.toLocalTime().toString()
             }
 
         // Prepare the datePickerDialog
-        val timePickerDialog = TimePickerDialog(this, timeSetListener,
-            selectedDate.hour, selectedDate.minute, true)
+        val timePickerDialog = TimePickerDialog(
+            this, timeSetListener,
+            selectedDate.hour, selectedDate.minute, true
+        )
 
         // Show the Dialog on the screen
         timePickerDialog.show()
 
     }
 
+    val checkBoxIdTime = mapOf<Int, Long>(
+        R.id.radio_notification_1h to Duration.ofHours(1).toMillis(),
+        R.id.radio_notification_5h to Duration.ofHours(5).toMillis(),
+        R.id.radio_notification_1d to Duration.ofDays(1).toMillis(),
+        R.id.radio_notification_3d to Duration.ofDays(3).toMillis()
+    )
+
+    private fun retrieveNotificationsTimes(): ArrayList<Long> {
+        var res = ArrayList<Long>()
+        for (x in checkBoxIdTime) {
+            if (findViewById<CheckBox>(x.key).isChecked) res.add(x.value)
+        }
+        return res
```

Comment:
This could probably be done functionally, with something like:
```Kotlin
return checkBoxIdTime
  .filter{ findViewById<CheckBox>(x.key).isChecked }
  .map{ it.value }
```
Otherwise you can keep it imperative, but in this case `res` should be a `val` and you should use an identifier that is more explicit than `x`.

Reply:
ok